### PR TITLE
Guide page for testing - first draft

### PR DESCRIPTION
**Pull request by sazzer**
I can't promise it's the best written documentation page, but it's a starting point for people to read to understand how they can test their code that makes use of React Query :)

Cheers

### CODE REVIEW

**Comment by tannerlinsley** [comment]
It looks great! Maybe we could use the isLoading/isSuccess from the query instead of waiting on the nock status?

**Comment by sazzer** [comment]
Hi there,

To be honest, the code I've got here was lifted almost as-is from my own
project, and because I'm using *Suspense* I didn't make use fo the
isLoading/isSuccess stuff.
I'll have a look at re-working it to use that for you tomorrow though.

Cheers

File: docs/src/pages/docs/guides/testing.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+const expectation = nock('http://example.com')
+  .get('/api/data')
+  .reply(200, {
+    answer: 42
+  });
+
+const { result, waitFor } = renderHook(() => useFetchData(), { wrapper });
+
+await waitFor(() => {
+  return expectation.isDone();
```
Comment (tannerlinsley) [inline docs/src/pages/docs/guides/testing.md:82]:
```suggestion
  return result.current.isSuccess();
```

**Review by tannerlinsley — CHANGES_REQUESTED** [review]

**Review by tannerlinsley — APPROVED** [review]

**Pull request merged**